rskj-core/src/test/java/co/rsk/db/IndexedBlockStoreTest.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2017 RSK Labs Ltd.
+ * (derived from ethereumJ library, Copyright (c) 2016 <ether.camp>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.db;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import co.rsk.core.BlockDifficulty;
+import co.rsk.crypto.Keccak256;
+import co.rsk.net.BlockStoreCache;
+import co.rsk.remasc.Sibling;
+import co.rsk.util.MaxSizeHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.ethereum.core.Block;
+import org.ethereum.core.BlockHeader;
+import org.ethereum.datasource.KeyValueDataSource;
+import org.ethereum.db.BlockStoreEncoder;
+import org.ethereum.db.IndexedBlockStore;
+import org.ethereum.db.IndexedBlockStore.BlockInfo;
+import org.junit.Before;
+import org.junit.Test;
+import org.mapdb.DB;
+
+
+public class IndexedBlockStoreTest {
```

Comment:
Tests look really good! Finally we start having _unit_ tests. I would suggest leaving out the javadocs and making sure the test names are sufficiently descriptive (which I think they are)

Reply:
I find it nice to understand tests such as "fails with this input" without internal knowledge of the class and cannot find a sufficiently descriptive name for that. I'll remove the ones I find redundant.